monai/handlers/classification_saver.py

```diff
@@ -0,0 +1,91 @@
+            overwrite (bool): whether to overwriting existing CSV file content. If we are not overwriting,
+                then we check if the results have been previously saved, and load them to the prediction_dict.
+            batch_transform (Callable): a callable that is used to transform the
+                ignite.engine.batch into expected format to extract the meta_data dictionary.
+            output_transform (Callable): a callable that is used to transform the
+                ignite.engine.output into the form expected model prediction data.
+                The first dimension of this transform's output will be treated as the
+                batch dimension. Each item in the batch will be saved individually.
+            name (str): identifier of logging.logger to use, defaulting to `engine.logger`.
+
+        """
+        self.output_dir = output_dir
+        self._prediction_dict = {}
+        self._preds_filepath = os.path.join(output_dir, 'predictions.csv')
+        if not overwrite:
+            if os.path.exists(self._preds_filepath):
+                with open(self._path, 'r') as f:
+                    reader = csv.reader(f)
+                    for row in reader:
+                        self._prediction_dict[row[0]] = np.array(row[1:]).astype(np.float32)
+
+        self.batch_transform = batch_transform
+        self.output_transform = output_transform
+
+        self.logger = None if name is None else logging.getLogger(name)
+
+    def attach(self, engine):
+        if self.logger is None:
+            self.logger = engine.logger
+        return engine.add_event_handler(Events.ITERATION_COMPLETED, self)
+
+    def finalize(self):
+        """
+        Writes the prediction dict to a csv
+
+        """
+        if not os.path.exists(self.output_dir):
+            os.makedirs(self.output_dir)
+        with open(self._preds_filepath, 'w') as f:
+            for k, v in sorted(self._prediction_dict.items()):
+                f.write(k)
+                for result in v.flatten():
+                    f.write("," + str(result))
+                f.write("\n")
+        self.logger.info('saved classification predictions into: {}'.format(self._preds_filepath))
+
+    def __call__(self, engine):
+        """
+        This method assumes self.batch_transform will extract Metadata from the input batch.
+        Metadata should have the following keys:
+
+            - ``'filename_or_obj'`` -- save the prediction corresponding to file name.
+
+        """
+        meta_data = self.batch_transform(engine.state.batch)
+        filenames = meta_data['filename_or_obj']
+
+        engine_output = self.output_transform(engine.state.output)
+        for batch_id, filename in enumerate(filenames):  # save a batch of files
+            output = engine_output[batch_id]
+            if isinstance(output, torch.Tensor):
+                output = output.detach().cpu().numpy()
+            self._prediction_dict[filename] = output
```

Comment:
this potentially creates inconsistency with respect to line 48 of this file (which has data type conversion)?